### ISSUE TITLE
padding-line-between-statements rules does not work as expected when mixed with case, default and return

### ISSUE THREAD

**Issue by ghs** [eslint/eslint#13561]
**Tell us about your environment**

* **ESLint Version:**
7.6.0
* **Node Version:**
12.13.0
* **npm Version:**
6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
"padding-line-between-statements": [
            "error",
            {
                "blankLine": "always",
                "prev": ["case", "default"],
                "next": ["*"]
            }
        ],
```
OR
```js
"padding-line-between-statements": [
            "error",
            {
                "blankLine": "always",
                "prev": ["*"],
                "next": ["return"]
            }
        ],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
switch (type) {
        case 'abc':
            return 1;
        case 'xyz':
            return 2;
        case 'qwerty':
            return 3;
        default:
            return 4;
        }
```

**What did you expect to happen?**
Get an error: "Expected blank line after this statement.eslint padding-line-between-statements" for **case** and **default**
OR
"Expected blank line before this statement.eslint padding-line-between-statements" for **return**

**What actually happened? Please include the actual, raw output from ESLint.**
When using "prev": ["case", "default"] and "next": ["*"], the error is for blank line before the case and default. When using "prev": ["*"] and "next": ["return"], it says nothing.

**Are you willing to submit a pull request to fix this bug?**
Not sure if would have the time at this moment :(

**Comment by mdjermanovic** [comment]
Hi @ghs, thanks for the issue!

I can confirm from [test cases](https://github.com/eslint/eslint/blob/488d1595aef43c4d52cccdb2c97977884f0375a8/tests/lib/rules/padding-line-between-statements.js#L600) that this works as expected.

`"case"` specifier applies to the whole `case` clause.

In your example:

```js
{
    "blankLine": "always",
    "prev": ["case", "default"],
    "next": ["*"]
}
```

This configuration means that there should be a blank line between case clauses:

```js
switch (type) {
    case 'abc':
        return 1;
    
    case 'xyz':
        return 2;
    
    case 'qwerty':
        return 3;
    
    default:
        return 4;
}
```

`"return"` has no effect here because there are no other statements around the `return` statements.

We should probably change the documentation to be more specific, namely `"case" is case labels.` should be `"case" is case clauses.` (and the same for `"default"`).

**Comment by ghs** [comment]
Great! Thanks for the clarification :)
